### PR TITLE
Add support to aggregated data for both error bar and error band

### PR DESCRIPTION
**Pull request by chanwutk**

### CODE REVIEW

**Comment by domoritz** [comment]
Could you add some tests to cover https://codecov.io/gh/vega/vega-lite/compare/536053c4bb0d84a7bf4cbd8fd50a32ded577410f...70c091bd0f34441af02aa2086e88b224f15991a7/src/src/compositemark/common.ts?

**Comment by chanwutk** [comment]
Should I merge this back to the error band #3820 branch?

File: src/compositemark/common.ts [hunk 1]
```diff
@@ -90,14 +90,17 @@ export function compositeMarkContinuousAxis<M extends CompositeMark>(
   const {mark: mark, encoding: encoding, projection: _p, ..._outerSpec} = spec;
 
   let continuousAxisChannelDef: PositionFieldDef<string>;
+  let continuousAxisChannelDef2: PositionFieldDef<string>;
   let continuousAxis: 'x' | 'y';
 
   if (orient === 'vertical') {
     continuousAxis = 'y';
     continuousAxisChannelDef = encoding.y as FieldDef<string>; // Safe to cast because if y is not continuous fielddef, the orient would not be vertical.
+    continuousAxisChannelDef2 = (encoding.y2) ? encoding.y2 as FieldDef<string> : undefined;
```
Comment (domoritz) [inline src/compositemark/common.ts]: why add () around this? Can you just write `encoding.y2 as FieldDef<string>`?

File: src/compositemark/common.ts [hunk 2]
```diff
@@ -90,14 +90,17 @@ export function compositeMarkContinuousAxis<M extends CompositeMark>(
...
   } else {
     continuousAxis = 'x';
     continuousAxisChannelDef = encoding.x as FieldDef<string>; // Safe to cast because if x is not continuous fielddef, the orient would not be horizontal.
+    continuousAxisChannelDef2 = (encoding.x2) ? encoding.x2 as FieldDef<string> : undefined;
```
Comment (domoritz) [inline src/compositemark/common.ts]: same as above

File: src/compositemark/common.ts [hunk 3]
```diff
@@ -108,8 +111,17 @@ export function compositeMarkContinuousAxis<M extends CompositeMark>(
     continuousAxisChannelDef = continuousAxisWithoutAggregate;
   }
 
+  if (continuousAxisChannelDef2 && continuousAxisChannelDef2.aggregate) {
+    const {aggregate, ...continuousAxisWithoutAggregate2} = continuousAxisChannelDef2;
+    if (aggregate !== compositeMark) {
+      log.warn(`Continuous axis should not have customized aggregation function ${aggregate}`);
```
Comment (domoritz) [inline src/compositemark/common.ts]: Can you explain why?

File: test/compositemark/errorbar.test.ts [hunk 4]
```diff
@@ -572,3 +572,210 @@ describe('normalizeErrorBar', () => {
...
+      },
+      "mark": "errorbar",
+      "encoding": {
+        "x": {"field": "age", "type": "ordinal"},
+        "y": {"field": "people", "type": "quantitative"},
+        "y2": {"field": "people2", "type": "quantitative", "aggregate": "min"}
+      }
+    }, defaultConfig);
+
+    assert.equal(localLogger.warns[0], 'Continuous axis should not have customized aggregation function min; errorbar already agregates the axis.');
```
Comment (domoritz) [inline test/compositemark/errorbar.test.ts]: Get the message from `log.ts`

Comment (chanwutk) [reply test/compositemark/errorbar.test.ts]: Fixed it.

**Comment by chanwutk** [comment]
@domoritz @kanitw Since we do not have center point, `iqr`and `ci` will stay the same no matter what is the `center`, and `stderr` and `stdev` are very unusual to be used with median. Should we take out the center property?

**Comment by kanitw** [comment]
> @domoritz @kanitw Since we do not have center point, iqrand ci will stay the same no matter what is the center, and stderr and stdev are very unusual to be used with median. Should we take out the center property?

I don't understand why there is no center point any more.  

I mean the center point won't be visualized, but we should still support error bar where users have raw data as input and specify center and extent.

**Comment by kanitw** [comment]
Why is documentation for error bar merged into this branch?

**Comment by chanwutk** [comment]
I thought it would be easier for rebasing. I forgot that it would be harder for you to review everything together.

**Comment by kanitw** [comment]
Manually merged to avoid rebasing the docs branch.

**Pull request merged**